solana/write-account/src/lib.rs

```diff
@@ -11,28 +16,47 @@ solana_program::entrypoint!(process_instruction);
 /// The first byte of the `instruction` determines operation to perform.  Format
 /// of the instruction and required accounts depend on that.
 ///
+/// # Create PDA Account
```

Comment:
Do we need a separate create command? I was imagining that write would create the account if it doesn’t exist. Similarly, write would resize it if it’s too small. I think it would be a simpler interface and require fewer smart contract calls.

Reply:
well the thing is, we cannot create the account and use it in the same method. So the account has to be created prior to calling the `write` method.

Reply:
Interesting.  How does Anchor do `init_if_needed`? I was under the impression that if the account doesn’t exist, it can be created and then operated on.

Reply:
even i thought the same, but looks like it creates an instruction and executes it before we call the method.